### PR TITLE
Refactor cicd pipeline

### PR DESCRIPTION
**Pull request by matthewfala**
This refactor allows for the sync tasks to be located on both production accounts and other accounts like the scanner account.

Also it serves as an abstraction for the CICD pipeline, to allow for the setup to be tested on test accounts.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: buildspec_load_test.yml [hunk 1]
```diff
@@ -61,17 +63,21 @@ phases:
...
       - export AWS_ACCESS_KEY_ID=""
       - export AWS_SECRET_ACCESS_KEY=""
       - export AWS_SESSION_TOKEN=""
       # Run load tests on corresponding platform
-      - python ./load_tests/load_test.py ${PLATFORM}
-    finally:
-      # Clear up testing resources
-      - python ./load_tests/load_test.py delete_testing_resources
+      - |
+        for i in {1..2}; do
```
Comment (weijianz-opc) [inline buildspec_load_test.yml:72]: should we make this `2` as a configure retry parameter ?

File: buildspec_load_test.yml [hunk 2]
```diff
@@ -61,17 +63,21 @@ phases:
...
+          python ./load_tests/load_test.py ${PLATFORM} && break \
+          || (
+            if [ $i == 2 ]; then
```
Comment (weijianz-opc) [inline buildspec_load_test.yml:75]: same `2` here.

File: buildspec_load_test_create_infra_ecs.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    CW_THROUGHPUT_LIST: '["1m", "2m", "3m"]'
+    TESTING_RESOURCES_STACK_NAME: 'load-test-fluent-bit-testing-resources'
+    LOG_STORAGE_STACK_NAME: 'load-test-fluent-bit-log-storage'
+    EKS_CLUSTER_NAME: 'load-test-fluent-bit-eks-cluster'
+    PREFIX: 'load-test-fluent-bit-'
+
+phases:
+  install:
+    runtime-versions:
+      golang: 1.16
```
Comment (weijianz-opc) [inline buildspec_load_test_create_infra_ecs.yml:15]: is there 1.x ?? reduce future upgrade effort >

File: load_tests/load_test.py [hunk 4]
```diff
@@ -18,8 +18,8 @@
 TESTING_RESOURCES_STACK_NAME = os.environ['TESTING_RESOURCES_STACK_NAME']
 PREFIX = os.environ['PREFIX']
 EKS_CLUSTER_NAME = os.environ['EKS_CLUSTER_NAME']
-LOGGER_RUN_TIME_IN_SECOND = 600
-BUFFER_TIME_IN_SECOND = 1000
+LOGGER_RUN_TIME_IN_SECOND = int(os.getenv("LOAD_TEST_RUN_TIME_IN_SECONDS", "600"))
```
Comment (weijianz-opc) [inline load_tests/load_test.py:21]: should we keep above ??  and do soemthing like 
```
DEFAULT_LOGGER_RUN_TIME_IN_SECOND = 600
LOGGER_RUN_TIME_IN_SECOND = int(os.getenv("LOAD_TEST_RUN_TIME_IN_SECONDS", DEFAULT_LOGGER_RUN_TIME_IN_SECOND))
``

File: load_tests/load_test.py [hunk 5]
```diff
@@ -436,22 +441,36 @@ def publish_fluent_config_s3(input_logger):
...
+        if os.getenv('CW_LOG_GROUP_NAME') != None:
+            client = session.client('logs')
+            response = client.describe_log_streams(
+                logGroupName=os.getenv('CW_LOG_GROUP_NAME')
+            )
+            for stream in response["logStreams"]:
+                client.delete_log_stream(
+                    logGroupName=os.environ['CW_LOG_GROUP_NAME'],
+                    logStreamName=stream["logStreamName"]
+                )
```
Comment (weijianz-opc) [inline load_tests/load_test.py:460]: should we pull this out into a separate method ?

File: load_tests/load_test.py [hunk 6]
```diff
@@ -436,22 +441,36 @@ def publish_fluent_config_s3(input_logger):
...
+            bucket.objects.all().delete()
+
+    else:
+        if os.getenv('S3_BUCKET_NAME') != None:
+            s3 = session.resource('s3')
+            bucket = s3.Bucket(os.environ['S3_BUCKET_NAME'])
+            bucket.objects.filter(Prefix=(output.lower() + "-test/")).delete()
+
 
 def generate_daemonset_config(throughput):
```
Comment (weijianz-opc) [inline load_tests/load_test.py:475]: consider break this method into smaller one for readability  line 532 - 609 , it too long to read :)

File: buildspec_load_test.yml (hunk 1 above)
Comment (PettitWesley) [inline buildspec_load_test.yml:72]: Why add retries like this? IIRC I added retries in the python code already, which I think is better than using bash.

File: buildspec_publish_ecr.yml [hunk 7]
```diff
@@ -36,7 +36,7 @@ phases:
         fi
 
       # Push the image to ECR
-      - './scripts/publish.sh cicd-publish ${REGION_TO_PUSH}'
+      - './scripts/publish.sh cicd-publish private-ecr'
```
Comment (PettitWesley) [inline buildspec_publish_ecr.yml:39]: why make this change?

File: buildspec_publish_ssm.yml [hunk 8]
```diff
@@ -11,7 +11,7 @@ phases:
     commands:
       # Enforce STS regional endpoints
       - export AWS_STS_REGIONAL_ENDPOINTS=regional
-      - './scripts/publish.sh cicd-publish-ssm ${AWS_REGION}'
+      - './scripts/publish.sh cicd-publish-ssm'
```
Comment (PettitWesley) [inline buildspec_publish_ssm.yml:14]: Why?

**Pull request closed without merging**